### ISSUE TITLE
When subtitle changed with "NextSubtitle" action it takes 20-30 seconds to display the new one

### ISSUE THREAD

**Issue by tonymontanadev** [xbmc/xbmc#15272]
### Describe the bug
Here is a clear and concise description of what the problem is:

I have a custom keymap and the yellow button on tv's remote switches the subtitle language to the next one. When I change the subtitle language with the yellow button, while the video is playing it takes 20-30seconds to start to display the new language on the screen. If I seek back and forth the subtitle starts to show instantly on the screen.

```
<keymap>
    <global>
        <remote>
            <red>ContextMenu</red>
        </remote>
    </global>
<FullscreenVideo>
    <remote>
        <red>OSD</red>
        <green>ShowSubtitles</green>
        <yellow>NextSubtitle</yellow>
        <blue>subtitledelay</blue>
    </remote>
</FullscreenVideo>
</keymap>
```

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

New subtitle should show instantly or start to show with the very next sentence.

## Actual Behavior

Takes to much time to display the subtitle on the screen.

### To Reproduce

1. Play a video which has several subtitle tracks
2. Switch the tracks with a single button, not from the menu

### Debuglog

The debug log can be found here: http://ix.io/1ywQ

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] Linux
 - [ ] OSX
 - [x] Raspberry-Pi
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: LibreELEC (Milhouse): devel-20190116042931-#0115-g3565115
 - Kodi version: Kodi (18.0-RC5 Git:3baca5a)

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by HitcherUK** [comment]
Have you tried using the video calibration option to check the position?

**Comment by tonymontanadev** [comment]
The subtitle position is fine. I think I found the cause. If I switch subtitle language on the interface by going into the options -> subtitle -> pick the language, the subtitle shows fine.

I have a custom keymap and the yellow button on tv's remote switches the subtitle language to the next one. When I switch with it, I have to stop the video and start again to make the subtitle visible. I remember it worked few months ago.

```
<keymap>
    <global>
        <remote>
            <red>ContextMenu</red>
        </remote>
    </global>
<FullscreenVideo>
    <remote>
        <red>OSD</red>
        <green>ShowSubtitles</green>
        <yellow>NextSubtitle</yellow>
        <blue>subtitledelay</blue>
    </remote>
</FullscreenVideo>
</keymap>
```

**Comment by DaVukovic** [comment]
You have banned add-ons/repositories installed. Please remove them and submit a new clean debuglog.
This issue will be marked as invalid for the time being.

Thanks

**Comment by tonymontanadev** [comment]
I deactivated the addons and also removed the repository. New log: http://ix.io/1ywQ

I did some testing and I think the problem is that when I change the subtitle language with the yellow button, while the video is playing it takes 20-30seconds to start to display the new language on the screen. If I seek back and forth the subtitle starts to show instantly on the screen.

This 20-30 seconds give me the idea that the subtitle is not working.
